core/tests/unit/api_core/helpers/test_general_helpers.py

```diff
@@ -0,0 +1,43 @@
+# Copyright 2017, Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import functools
+
+from google.api.core.helpers import general_helpers
+
+
+def test_wraps_normal_func():
+
+    def func():
+        return 42
+
+    @general_helpers.wraps(func)
+    def replacement():
+        return func()
+
+    assert replacement() == 42
+
+
+def test_wraps_partial():
```

Comment:
It seems this would be the example:

```python
>>> import sys
>>> sys.version
'2.7.14 (default, Sep 19 2017, 20:26:00) \n[GCC 5.4.0 20160609]'
>>>
>>> def func():
...     return 42
...
>>>
>>> import functools
>>> partial = functools.partial(func)
>>>
>>> import six
>>> @six.wraps(partial)
... def replacement():
...     return partial()
... 
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/dhermes/.pyenv/versions/2.7.14/lib/python2.7/site-packages/six.py", line 811, in wrapper
    f = functools.wraps(wrapped, assigned, updated)(f)
  File "/home/dhermes/.pyenv/versions/2.7.14/lib/python2.7/functools.py", line 33, in update_wrapper
    setattr(wrapper, attr, getattr(wrapped, attr))
AttributeError: 'functools.partial' object has no attribute '__module__'
```

Reply:
Yep, this showed up in #4148